gulpfile.js

```diff
@@ -1,210 +1,182 @@
-
-}
-
-function wipeDist(done) {
-	var dir = path.join(__dirname, '/dist/');
-	console.log('dir', dir);
-	wipe(dir, done);
+/** 
+ * Builds files using webpack.
+ */
+const buildWebpack = () => {
+	return webpack(webpackFilesConfig);
 }
 
+/**
+ * Launches the application.
+ * 
+ * @param {string} env The build environment. 
+ */
+const launchApplication = env => {
+	ON_DEATH((signal, err) => {
+		exec("taskkill /F /IM openfin.* /T", (err, stdout, stderr) => {
+			// Only write the error to console if there is one and it is something other than process not found.
+			if (err && err !== 'The process "openfin.*" not found.') {
+				console.error(errorOutColor(err));
+			}
 
-function wipe(dir, cb) {
-	del(dir, { force: true }).then(function () {
-		if (cb) {
-			cb();
-		}
-	}).catch(function (err) {
-		console.error(err);
+			process.exit();
+		});
 	});
-}
-
-function buildSass(done) {
-	done();
-	return gulp.src([
-		path.join(__dirname, '/src/components/**/**/*.scss'),
-		//compiles sass down to finsemble.css
-		path.join(__dirname, '/src/components/assets/*.scss')
-	])
-		.pipe(sass().on('error', sass.logError))
-		.pipe(gulp.dest(path.join(__dirname, '/dist/components/')));
-}
-
-function watchSass(done) {
-	watch(path.join(__dirname, '/src/components/assets/**/*'), {}, gulp.series(buildSass));
-	done();
-}
-
-function watchStatic() {
-	watch(path.join(__dirname, '/src/**/*.css'), { ignoreInitial: true }).pipe(gulp.dest(path.join(__dirname, '/dist/')));
-	return watch(path.join(__dirname, '/src/**/*.html'), { ignoreInitial: true }).pipe(gulp.dest(path.join(__dirname, '/dist/')));
-}
-
-function wipedist(done) {
-	if (directoryExists(path.join(__dirname, "/dist/"))) {
-		wipe(path.join(__dirname, '/dist/'), done);
-	} else {
-		done();
-	}
-}
 
-function handleWebpackStdOut(data, done) {
-	let notAnError = !data.includes('build failed');
-	if (notAnError) {
-		console.log(webpackOutColor(data));
-	}
-
-	if (data.includes('webpack is watching')) {
-
-		if (initialBuildFinished && notAnError) {
-			// buildComplete();
-		} else if (!initialBuildFinished) {
-			done();
-		}
-	}
-}
-
-function webpackComponents(done) {
-	var webpack_config = require('./build/webpack/webpack.files.js')
-	webpack(webpack_config, function (err, stats) {
-		done();
-	})
-}
-function webpackServices(done) {
-	var webpack_config = require('./build/webpack/webpack.services.js')
-	webpack(webpack_config, function (err, stats) {
-		done();
-	})
-}
-function launchOpenfin(env) {
-	const { exec } = require('child_process');
-	var OFF_DEATH = ON_DEATH(function (signal, err) {
-		exec('taskkill /F /IM openfin.* /T', (err, stdout, stderr) => {
-			if (err) {
-				console.error(err)
-				this.process.exit();
-				// node couldn't execute the command
-				return;
-			}
-			this.process.exit();
+	return launcher
+		.launchOpenFin({
+			configPath: startupConfig[env].serverConfig
+		})
+		.then(() => {
+			// OpenFin has closed so exit gulpfile
+			process.exit();
```

Comment:
wow. does this solve ctrl+c?????

Reply:
I think so. If you close finsemble in dev mode and OF closes, it kills the gulpfile execution.